source/math/Quaternion.ooc

```diff
@@ -260,6 +262,80 @@ Quaternion: cover {
 		}
 		result
 	}
+	weightedMeanOfQuaternions: static func (quaternions: VectorList<This>, weights: Float[]) -> This {
+		// Implementation of the QUEST algorithm. For variable meanings, see original publication:
+		// M.D. Shuster and S.D. Oh, "Three-Axis Attitude Determination from Vector Observations", 1981
+		// [http://www.malcolmdshuster.com/Pub_1981a_J_TRIAD-QUEST_scan.pdf]
+		// TODO: Fix singularity problem when the angle is close to PI, using sequential rotations
+		vectorCount := 3 * quaternions count
+		referenceVectors := FloatMatrix new(vectorCount, 3)
+		observationVectors := FloatMatrix new(vectorCount, 3)
+		This _createVectorMeasurementsForQuest(quaternions, referenceVectors, observationVectors)
+		This _normalizeFloatArray(weights, quaternions count)
+
+		B := FloatMatrix new(3, 3)
+		w := FloatMatrix new(1, 3)
+		v := FloatMatrix new(1, 3)
+		for (n in 0 .. vectorCount) {
+			for (i in 0 .. referenceVectors dimensions height) {
+				v set(0, i, referenceVectors get(n, i))
+				w set(0, i, observationVectors get(n, i))
+			}
+			B += weights[n / 3] / 3.0f * w * v transpose()
+		}
+		S := B + B transpose()
+		w free()
+		v free()
+
+		temporaryZ := FloatPoint3D new()
+		for (n in 0 .. vectorCount)
+			temporaryZ += weights[n / 3] / 3.0f * FloatPoint3D new(observationVectors get(n, 0), observationVectors get(n, 1), observationVectors get(n, 2)) vectorProduct(FloatPoint3D new(referenceVectors get(n, 0), referenceVectors get(n, 1), referenceVectors get(n, 2)))
```

Comment:
I was discussing it but haven't done it, but I can look into it. Is there a way of defining the []-operator to take two arguments? Otherwise, how should it be done?